substrate-archive/src/sql_block_builder.rs

```diff
@@ -23,85 +23,55 @@ use std::marker::PhantomData;
 use codec::{Decode, Encode};
 use sp_runtime::{
 	generic::SignedBlock,
-	traits::{Block as BlockT, DigestFor, Header as HeaderT},
+	traits::{Block as BlockT, Header as HeaderT},
 };
 
 use substrate_archive_common::{models::BlockModel, types, Error};
 
-struct GenericParts<B: BlockT>(
-	<B::Header as HeaderT>::Hash,
-	<B::Header as HeaderT>::Hash,
-	<B::Header as HeaderT>::Hash,
-);
-
-pub struct BlockBuilder<B: BlockT> {
+pub struct SqlBlockBuilder<B: BlockT> {
 	_marker: PhantomData<B>,
 }
 
-impl<'a, B: BlockT> BlockBuilder<B> {
+impl<'a, B: BlockT> SqlBlockBuilder<B> {
 	/// With a vector of BlockModel
 	pub fn with_vec(blocks: Vec<BlockModel>) -> Result<Vec<types::Block<B>>, Error> {
 		blocks
 			.into_iter()
 			.map(|b| {
-				let (b, s) = Self::with_single(b)?;
-				let b = SignedBlock { block: b, justification: None };
-				Ok(types::Block::new(b, s))
+				let (block, spec) = Self::with_single(b)?;
+				let block = SignedBlock { block, justification: None };
```

Comment:
Not related to this PR, but just out of curiosity is there any reason we don't store a justification? From my understanding there is at least 1 block with a stored justification per session.

Reply:
Here, the justification is unnecessary since if we already have the block in the database, a justification isn't needed for re-executing the block. Justification is used for consensus purposes/proving that X block is indeed finalized because some number of people/validators voted on it I think (my understanding there is rudimentary I think). Since archive doesn't do anything with consensus we can safely ignore this

However, we actually don't store justifications at all in the database. We have the justification on the block before it's inserted into the database, there's just not justification column. I can't remember if there was a reason behind this, but I think it would be good to support storing justifications.

Reply:
Created an issue for this: https://github.com/paritytech/substrate-archive/issues/147